packages/utils/src/object.ts

```diff
@@ -301,84 +301,83 @@ function makeSerializable<T>(value: T, key?: any): T | string {
 }
 
 /**
- * Walks an object to perform a normalization on it
+ * normalize()
  *
- * @param key of object that's walked in current iteration
- * @param value object to be walked
- * @param depth Optional number indicating how deep should walking be performed
- * @param memo Optional Memo class handling decycling
+ * - Creates a copy to prevent original input mutation
+ * - Skip non-enumerablers
+ * - Calls `toJSON` if implemented
+ * - Removes circular references
+ * - Translates non-serializeable values (undefined/NaN/Functions) to serializable format
+ * - Translates known global objects/Classes to a string representations
+ * - Takes care of Error objects serialization
+ * - Optionally limit depth of final output
  */
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export function walk(key: string, value: any, depth: number = +Infinity, memo: MemoFunc = memoBuilder()): any {
-  const [memoize, unmemoize] = memo;
+export function normalize(input: unknown, maxDepth: number = +Infinity, maxEdges: number = 10_000): any {
```

Comment:
wait `10_000` works :0

Does it get transpiled properly?

Reply:
We'll probably have to make `maxEdges` configurable like `maxDepth`

Reply:
Yep, all gets transpiled by tsc. 

Make `maxEdges` configurable, or remove the default value?

Reply:
we should keep the default, but something similar to https://github.com/getsentry/sentry-javascript/blob/845aada785a956e6348d87da9ff183aec24b5add/packages/types/src/options.ts#L102

Also I wonder if this change will be disruptive to existing customers in any way. Any reason to choose `10_000`?

Reply:
10k is just a rough guestimate for what will likely not be noticed by customers but still limit crazy large objects.

I think the alternative of limiting number of properties/elements per object/array might be less disruptive and output more useful serialised output. The issue I see with this edges approach is that it's hard to pick a `maxEdges` value and when that number is reached, nothing more is outputted. This includes skipping properties further up the object tree.

Reply:
> The issue I see with this edges approach is that it's hard to pick a maxEdges value and when that number is reached, nothing more is outputted

Yeah good observation. I think `limiting number of properties/elements per object/array` is also much more intuitive to understand - and adjusting it will be much more obvious.